internal/config/configdomain/use_message.go

```diff
@@ -0,0 +1,98 @@
+package configdomain
+
+import (
+	"github.com/git-town/git-town/v17/internal/git/gitdomain"
+	. "github.com/git-town/git-town/v17/pkg/prelude"
+)
+
+// UseMessage indicates how to set commit message.
+// The following variants are supported:
+// - EditDefaultMessage
+// - UseCustomMessage(CommitMessage)
+// - UseDefaultMessage
+//
+// See `--message`, `--edit` and `--no-edit` flags in `git commit` and `git merge`.
+type UseMessage struct {
+	customMessage      Option[gitdomain.CommitMessage]
+	editDefaultMessage bool
+}
+
+// EditDefaultMessage indicates that the default message should be edited.
+func EditDefaultMessage() UseMessage {
+	return UseMessage{
+		customMessage:      None[gitdomain.CommitMessage](),
+		editDefaultMessage: true,
+	}
+}
+
+// UseCustomMessage indicates that the custom message should be used as is without opening an editor.
+func UseCustomMessage(message gitdomain.CommitMessage) UseMessage {
+	return UseMessage{
+		customMessage:      Some(message),
+		editDefaultMessage: false, // This value is unused.
+	}
+}
+
+// UseCustomMessageOr returns UseCustomMessage(message) if message is Some, or other.
+func UseCustomMessageOr(message Option[gitdomain.CommitMessage], other UseMessage) UseMessage {
+	if m, has := message.Get(); has {
+		return UseCustomMessage(m)
+	}
+	return other
+}
+
+// UseDefaultMessage indicates that the default message should be used as is without opening an editor.
+func UseDefaultMessage() UseMessage {
+	return UseMessage{
+		customMessage:      None[gitdomain.CommitMessage](),
+		editDefaultMessage: false,
+	}
+}
+
+// UseMessageWithFallbackToDefault returns UseMessage based on an optional message and fallbackToDefault.
+// - If message is Some(...) then this is equivalent to UseCustomMessage(...).
+// - If message is None and fallbackToDefault then this is equivalent to UseDefaultMessage().
+// - If message is None and !fallbackToDefault then this is equivalent to EditDefaultMessage().
+func UseMessageWithFallbackToDefault(message Option[gitdomain.CommitMessage], fallbackToDefault bool) UseMessage {
+	return UseMessage{
+		customMessage:      message,
+		editDefaultMessage: !fallbackToDefault,
+	}
+}
+
+// GetCustomMessage returns a copy of the custom message and IsCustomMessage().
+func (self *UseMessage) GetCustomMessage() (gitdomain.CommitMessage, bool) {
```

Comment:
Not feeling strongly about this, but am curious whether this method could just return `Option[gitdomain.CommitMessage]`?

Reply:
While this looks similar, I disagree. If we just return the inner `customMessage` this starts to look like a direct access to the private state.

Instead the boolean here indicates whether the UseMessage instance is CustomMessage or not. And if it is logically it contains non-optional CommitMessage.

Reply:
Okay, thanks for the background. Makes sense.